core/src/banking_stage/transaction_scheduler/transaction_state.rs

```diff
@@ -1,53 +1,32 @@
 use crate::banking_stage::scheduler_messages::MaxAge;
 
-/// Simple wrapper type to tie a sanitized transaction to max age slot.
-pub(crate) struct SanitizedTransactionTTL<Tx> {
-    pub(crate) transaction: Tx,
-    pub(crate) max_age: MaxAge,
-}
-
 /// TransactionState is used to track the state of a transaction in the transaction scheduler
 /// and banking stage as a whole.
 ///
-/// There are two states a transaction can be in:
-///     1. `Unprocessed` - The transaction is available for scheduling.
-///     2. `Pending` - The transaction is currently scheduled or being processed.
-///
-/// Newly received transactions are initially in the `Unprocessed` state.
-/// When a transaction is scheduled, it is transitioned to the `Pending` state,
-///   using the `transition_to_pending` method.
-/// When a transaction finishes processing it may be retryable. If it is retryable,
-///   the transaction is transitioned back to the `Unprocessed` state using the
-///   `transition_to_unprocessed` method. If it is not retryable, the state should
-///   be dropped.
-///
-/// For performance, when a transaction is transitioned to the `Pending` state, the
-///   internal `SanitizedTransaction` is moved out of the `TransactionState` and sent
-///   to the appropriate thread for processing. This is done to avoid cloning the
-///  `SanitizedTransaction`.
-#[allow(clippy::large_enum_variant)]
-pub(crate) enum TransactionState<Tx> {
-    /// The transaction is available for scheduling.
-    Unprocessed {
-        transaction_ttl: SanitizedTransactionTTL<Tx>,
-        priority: u64,
-        cost: u64,
-    },
-    /// The transaction is currently scheduled or being processed.
-    Pending { priority: u64, cost: u64 },
-    /// Only used during transition.
-    Transitioning,
```

Comment:
Happy to see `Transitioning` is gone - always think transitioning between states itself is not a state.